### PR TITLE
#290 Sequentially Consistent Santa

### PR DESCRIPTION
**Pull request by oodler577**
Ready for review.

### CODE REVIEW

**Comment by oodler577** [comment]
slower going because I finding some bugs in the code, which is good but I need to fix them - https://github.com/oodler577/p5-Sub-Genius/tree/release-0.314005

**Comment by oodler577** [comment]
@oalders I have some `=begin text ... =end text` blocks because they're not Perl but should be verbatim; not sure if that's right. Happy to make it right though.

File: 2023/incoming/issue-290.pod [hunk 1]
```diff
@@ -0,0 +1,404 @@
...
+
+=back 
+
+The work plans are fed into the box using the DSL format that is then read
+and processed by an internal module called C<Sub::Genius>.  A module aptly
+named since its original author is certainly not the brightest bulb in the
+shed. But the module is a good example of the many useful tools that exist
+on the Internet.
+
+After being processed by C<Sub::Genius>, the plan is turned into a set of
```
Comment (oalders) [inline 2023/incoming/issue-290.pod:58]:
```suggestion
After being processed by L<Sub::Genius>, the plan is turned into a set of
```

File: 2023/incoming/issue-290.pod [hunk 2]
```diff
@@ -0,0 +1,404 @@
...
+my $sg = Sub::Genius->new(preplan => $plan)->init_plan;
+
+# print all sequential orderings of the steps described above
+while (my $preplan = $sg->next()) {
+  print qq{$preplan\n}; 
+}
+
+=end perl
+
+When run, this prints all 588 possible correct order of operations. A
```
Comment (oalders) [inline 2023/incoming/issue-290.pod:114]: "Orders of operation"?

File: 2023/incoming/issue-290.pod [hunk 3]
```diff
@@ -0,0 +1,404 @@
...
+sequential set of steps is I<correct> if it respects the partial ordering
+among dependent steps described in the plan above.
+
+For example, C<HandleBars> must always be installed before the
+C<RearBreaks>. But the C<Peddles> may be installed before or after the
+C<FrontBreaks>. And in this way all independent tasks may be interleaved,
```
Comment (oalders) [inline 2023/incoming/issue-290.pod:120]:
```suggestion
C<RearBrakes>. But the C<Peddles> may be installed before or after the
C<FrontBrakes>. And in this way all independent tasks may be interleaved,
```

File: 2023/incoming/issue-290.pod [hunk 4]
```diff
@@ -0,0 +1,404 @@
...
+the steps in a properly ordered way. But alas, C<Sub::Genius> is far more
+useful than that!
+
+It so happens that if the C<sub>s with the same names as the I<steps> of the
+plan are defined within the scope, then the plan may be actually executed. This
+allows for each step in the plan to have actions associated with them (even
+generating additional nested plans).
+
+Because creating scripts to be executed according to the plan is a tedious
+task, one of the useful tools C<Sub::Genius> provides is called C<stubby>.
```
Comment (oalders) [inline 2023/incoming/issue-290.pod:157]:
```suggestion
task, one of the useful tools L<Sub::Genius> provides is called C<stubby>.
```

File: 2023/incoming/issue-290.pod [hunk 5]
```diff
@@ -0,0 +1,404 @@
...
+
+use strict;
+use warnings;
+use Sub::Genius;
+
+my $plan = <<EOP;
+  ForgeBike
+  InitialPaint
+  (
+    ( ( RearWheel & Peddles & Gears ) ( RearBrakes & Chain))
```
Comment (oalders) [inline 2023/incoming/issue-290.pod:180]: I think we'll need to `s/Peddles/Pedals/` on the whole document.

**Review by oalders — CHANGES_REQUESTED** [review]
Looks great! Just needs a few edits.

File: 2023/incoming/issue-290.pod (hunk 5 above)
Comment (oodler577) [reply 2023/incoming/issue-290.pod:180]: I thought it looked funny. lol

**Review by oalders — APPROVED** [review]
Thanks, @oodler577!

**Pull request merged**